### PR TITLE
General settings migrate to redux

### PR DESCRIPTION
**Pull request by uldisrudzitis**

### CODE REVIEW

File: _api_app/app/Http/Controllers/SettingsController.php [hunk 1]
```diff
@@ -0,0 +1,27 @@
...
+
+    public function update(Request $request) {
+        $json = $request->json()->all();
+        $path_arr = explode('/', $json['path']);
+        $site = $path_arr[0];
+        $settings = new SiteSettings($site);
+        $path_arr = array_slice($path_arr, 1);
+
+        $res = $settings->saveValueByPath($json['path'], $json['value']);
+        // @@@:TODO: Replace this with something sensible, when migration to redux is done
```
Comment (vteivans) [inline _api_app/app/Http/Controllers/SettingsController.php]: how is this meant? What the fields `update` and `real` are supposed to contain?

File: _api_app/app/SiteSettings.php [hunk 2]
```diff
@@ -2,18 +2,73 @@
...
+    public function saveValueByPath($path, $value)
+    {
+        $settings = $this->get();
+        $path_arr = array_slice(explode('/', $path), 2);
+        $value = trim(urldecode($value));
+
+        $ret = array(
+            'site' => $this->SITE,
+            'path' => $path,
+            'value' => $value,
```
Comment (vteivans) [inline _api_app/app/SiteSettings.php]: Why do you return urlencoded value?

File: _api_app/app/Http/Controllers/SettingsController.php (hunk 1 above)
Comment (uldisrudzitis) [reply _api_app/app/Http/Controllers/SettingsController.php]: `update` is an updated value, `real` as a  transformed and formatted value for old Berta front end needs (url prefixes, date format etc).

Comment (vteivans) [reply _api_app/app/Http/Controllers/SettingsController.php]: So the idea is that the `update` field will contain some kind of new value when the migration to redux is done, but the `real` field will contain backward compatible value?

**Pull request merged**